lib/matplotlib/pyplot.py

```diff
@@ -249,10 +272,10 @@ def _get_backend_mod():
         # logic (which will (re)import pyplot and then call switch_backend if
         # we need to resolve the auto sentinel)
         switch_backend(rcParams._get("backend"))
-    return _backend_mod
+    return cast(matplotlib.backend_bases._Backend, _backend_mod)
```

Comment:
yeah, if it _returned_ the backend and was set back to the variable by `switch_backend`, it would narrow, but because it is side-effect behavior of a function call, the type checker is unaware of that connection. The cast is not ideal, but works, may also have accepted a `# type: ignore` comment for that reason (which has the advantage of no runtime method lookup (for a runtime no-op), but the disadvantage of potentially ignoring a bit more than you care to.

There are some kind of strange things that I haven't fully looked at in relation to `_backend_mod`, particularly a scoped class definition that occludes the variable, but then seems to refer to the variable form internally?

Kind of feels like it shouldn't work, and didn't really see where the class was instantiated immediately.

Reply:
Yea, it causes some weird issues due to re-using the name. I think I might have figured it out, but there's still a few complaints from `mypy` to figure out.